server/utils/utils.test.ts

```diff
@@ -414,94 +416,174 @@ describe('Get prisoner release date from Nomis', () => {
+    expect(isPassedArdOrCrd(licence, prisoner)).toBe(true)
+  })
 
-      expect(isPassedArdOrCrd(licence, prisoner)).toBe(false)
-    })
+  it('Should return false when legal status is immigration and conditionalReleaseDate is tomorrow', () => {
+    const licence = {
+      actualReleaseDate: undefined,
+      conditionalReleaseDate: format(addDays(new Date(), 1), 'dd/MM/yyyy'),
+    } as LicenceSummary
 
-    it('Should return false when legal status is immigration and no dates are present', () => {
-      const licence = {
-        actualReleaseDate: undefined,
-        conditionalReleaseDate: undefined,
-      } as LicenceSummary
+    const prisoner = { legalStatus: 'IMMIGRATION_DETAINEE' } as Prisoner
 
-      const prisoner = { legalStatus: 'IMMIGRATION_DETAINEE' } as Prisoner
+    expect(isPassedArdOrCrd(licence, prisoner)).toBe(false)
+  })
 
-      expect(isPassedArdOrCrd(licence, prisoner)).toBe(false)
-    })
+  it('Should return true when legal status is immigration and actualReleaseDate is present and in the past', () => {
+    const licence = {
+      actualReleaseDate: format(new Date(), 'dd/MM/yyyy'),
+      conditionalReleaseDate: format(addDays(new Date(), 1), 'dd/MM/yyyy'),
+    } as LicenceSummary
+
+    const prisoner = { legalStatus: 'IMMIGRATION_DETAINEE' } as Prisoner
+
+    expect(isPassedArdOrCrd(licence, prisoner)).toBe(true)
+  })
+
+  it('Should return false when legal status is immigration and actualReleaseDate is present and in the future', () => {
+    const licence = {
+      actualReleaseDate: format(addDays(new Date(), 1), 'dd/MM/yyyy'),
+      conditionalReleaseDate: format(new Date(), 'dd/MM/yyyy'),
+    } as LicenceSummary
+
+    const prisoner = { legalStatus: 'IMMIGRATION_DETAINEE' } as Prisoner
+
+    expect(isPassedArdOrCrd(licence, prisoner)).toBe(false)
+  })
+
+  it('Should return false when legal status is immigration and no dates are present', () => {
+    const licence = {
+      actualReleaseDate: undefined,
+      conditionalReleaseDate: undefined,
+    } as LicenceSummary
+
+    const prisoner = { legalStatus: 'IMMIGRATION_DETAINEE' } as Prisoner
+
+    expect(isPassedArdOrCrd(licence, prisoner)).toBe(false)
+  })
+})
+
+describe('releaseDateLabel', () => {
+  it('returns "CRD" if the release date from NOMIS is a CRD', () => {
+    const nomisRecord = {
+      conditionalReleaseDate: '2035-11-22',
+    } as Prisoner
+
+    expect(releaseDateLabel(null, nomisRecord)).toBe('CRD')
+  })
+
+  it('returns "Confirmed release date" if the release date from NOMIS is a confirmed release date', () => {
+    const nomisRecord = {
+      confirmedReleaseDate: '2035-10-22',
+    } as Prisoner
+
+    expect(releaseDateLabel(null, nomisRecord)).toBe('Confirmed release date')
+  })
+
+  it('returns "CRD" if the release date from the licence is a CRD, regardless of NOMIS record', () => {
+    const licence = {
+      conditionalReleaseDate: '2035-11-22',
```

Comment:
Think the date format is dd/MM/YYYY

also be wary that javascript new date interprets `MM/dd/YYYY`
<img width="501" alt="Screenshot 2023-02-03 at 08 18 05" src="https://user-images.githubusercontent.com/1517745/216548147-5931e442-430b-415f-8fd0-2db92022f90e.png">

Reply:
You're exactly right and I've added a step to avoid the `new Date` issue (and also to add consistency to date formatting if the date is somehow corrupted/invalid)

Reply:
should these tests be updated then?

Reply:
Yes... yes they should 😅 
I changed a handful, didn't realise I'd missed so many!